beacon-chain/sync/custody.go

```diff
@@ -103,12 +103,27 @@ func (s *Service) custodyGroupCount(context.Context) (uint64, error) {
 		return cfg.NumberOfCustodyGroups, nil
 	}
 
+	// Calculate validator custody requirements
 	validatorsCustodyRequirement, err := s.validatorsCustodyRequirement()
 	if err != nil {
 		return 0, errors.Wrap(err, "validators custody requirement")
 	}
 
-	return max(cfg.CustodyRequirement, validatorsCustodyRequirement), nil
+	effectiveCustodyRequirement := max(cfg.CustodyRequirement, validatorsCustodyRequirement)
+
+	// If we're not in semi-supernode mode, just use the effective requirement.
+	if !flags.Get().SemiSupernode {
+		return effectiveCustodyRequirement, nil
+	}
+
+	// Semi-supernode mode custodies the minimum custody groups required for reconstruction.
+	// This is future-proof and works correctly even if custody groups != columns.
+	semiSupernodeTarget := peerdas.MinimumCustodyGroupCountToReconstruct()
+	if effectiveCustodyRequirement <= semiSupernodeTarget {
+		return semiSupernodeTarget, nil
+	}
+
+	return effectiveCustodyRequirement, nil
```

Comment:
```suggestion
	return max(effectiveCustodyRequirement, semiSupernodeTarget), nil
```

Cf. Preston's modernisation PR